src/components/upload/FileUpload.js

```diff
@@ -40,43 +35,65 @@ const FileUpload = () => {
     const [state, setState] = useState("idle");
     const [projectFile, setProjectFile] = useState();
     const [projectName, setProjectName] = useState("");
+    const [open,setOpen] = React.useState(false);
+    const [enableButton] = useState("");
+    const navigate = useNavigate();
 
-    const onSubmit = () => {
-        setState("uploading");
+       const onSubmit = (e) => {
+        setOpen(true);
+        e.preventDefault();
         uploadProjectFile(projectName, projectFile)
-            .then(() => {
-                Notifications.notify("Project created.", "success")
-                setState("idle");
-            })
-            .catch((err) => {
-                Notifications.notify(`Failed to create project.\n${err}`, "error")
-                setState("idle");
-            })
+         if (projectName && projectFile) {
+            Notifications.notify("Project created.", "success")
+            setState("idle");
+            setOpen(true);
+            navigate("/generate", { replace: true });
```

Comment:
```suggestion
            setState("idle");
            uploadProjectFile(projectName, projectFile)
                .then(() => {
                    navigate("/generate", { replace: true });
                });
```
This one is a bit of a meany as you just learn programming. `uploadProjectFile` is not a function that behaves as other synchronous functions. If you want to ensure that `uploadProjectFile` has been finished you need to call `.then` on it's result as shown in this suggestion.